### PR TITLE
Add Decimal128 as primitive type.

### PR DESCRIPTION
**Pull request by hpoul**
fixes https://github.com/szokodiakos/typegoose/issues/114 by simply adding `Decimal128` as primitive type. Since it is perfectly supported by mongoose it seems to work.

### CODE REVIEW

File: src/test/index.test.ts [hunk 1]
```diff
@@ -10,6 +10,8 @@ import { Genders } from './enums/genders';
 import { Role } from './enums/role';
 import { initDatabase, closeDatabase } from './utils/mongoConnect';
 import { getClassForDocument } from '../utils';
+import {Decimal128} from 'bson';
+import {fail} from 'assert';
```
Comment (Ben305) [inline src/test/index.test.ts]: Just a cosmetic issue, but please add spaces after the opening and before the closing bracket.

File: src/test/models/car.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import * as mongoose from 'mongoose';
 
 import { Typegoose, prop, pre } from '../../typegoose';
+import {Decimal128} from 'bson';
```
Comment (Ben305) [inline src/test/models/car.ts]: Just a cosmetic issue, but please add spaces after the opening and before the closing bracket.

**Comment by hpoul** [comment]
@Ben305 sure, i've now added the whitespace (and enabled a tslint rule to check for it ;-) ) let me know if there is anything else. thanks.

**Comment by Ben305** [comment]
@hpoul Thanks for you contribution, released with v5.1.0 👍

**Pull request merged**